src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LongHopscotchSet.java

```diff
@@ -0,0 +1,681 @@
+            predecessorBucketIndex = getIndex(bucketToMoveIndex, nextOffset);
+            while ((nextOffset = getOffset(predecessorBucketIndex)) != 0 && nextOffset < toEmptyDistance) {
+                toEmptyDistance -= nextOffset;
+                predecessorBucketIndex = getIndex(predecessorBucketIndex, nextOffset);
+            }
+            status[predecessorBucketIndex] = (byte) toEmptyDistance;
+        }
+        if (nextOffset != 0) {
+            status[emptyBucketIndex] = (byte) (nextOffset - toEmptyDistance);
+        }
+        buckets[emptyBucketIndex] = buckets[bucketToMoveIndex];
+        buckets[bucketToMoveIndex] = 0;
+        status[bucketToMoveIndex] = 0;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void resize() {
+        if (buckets == null) {
+            throw new IllegalStateException("Someone must be doing something ugly with reflection -- I have no buckets.");
+        }
+        final int oldCapacity = capacity;
+        final int oldSize = size;
+        final long[] oldBuckets = buckets;
+        final byte[] oldStatus = status;
+
+        capacity = getLegalSizeAbove(capacity);
+        size = 0;
+        buckets = new long[capacity];
+        status = new byte[capacity];
+
+        try {
+            int idx = 0;
+            do {
+                final long entry = oldBuckets[idx];
+                if (!isNull(entry)) insert(getValue(entry), (t1, t2) -> false);
+            }
+            while ((idx = (idx + 127) % oldCapacity) != 0);
+        } catch (final IllegalStateException ise) {
+            capacity = oldCapacity;
+            size = oldSize;
+            buckets = oldBuckets;
+            status = oldStatus;
+            // this shouldn't happen except in the case of really bad hashCode implementations
+            throw new IllegalStateException("Hopscotching failed at load factor " + 1. * size / capacity + ", and resizing didn't help.");
+        }
+
+        if (size != oldSize) {
+            // this should never happen, period.
+            throw new IllegalStateException("Lost some elements during resizing.");
+        }
+    }
+
+    public boolean containsAll(final long[] vals) {
+        for (final long val : vals) {
+            if (!contains(val))
+                return false;
+        }
+        return true;
+    }
+
+    public void addAll(final long[] entryValues) {
+        for (final long val : entryValues) {
+            add(val);
+        }
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        final LongHopscotchSet that = (LongHopscotchSet) o;
+
+        if (size != that.size) return false;
+        final LongIterator itr = this.iterator();
+        while (itr.hasNext()) {
+            if (!that.contains(itr.next())) return false;
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
To meet the contract for hashCode and equals, this must be defined like it is in LargeLongHopscotchSet:  as a sum over the hashCodes of the elements.